sycl: update documentation (#2129)
oneapi-src/oneDNN: src/gpu/generic/sycl/README.md

# Supported Primitives

General limitations:

* Currently blocked formats are not supported by any implementation unless

>     explicitly listed

Author: Fixed, thanks.

* There's a limit of maximum 5 post-ops for the implementations
* The maximum supported size of any dimension of any input/output tensor of a
    primitive is `INT32_MAX`

## Batch Normalization

The implementation supports both forward and backward directions.

* Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`, `NC`
* Supported data types
    * Forward direction: `f32`, `bf16`, `f16`, `s8`
    * Backward direction: `f32`, `bf16`, `f16`

## Binary

* Supported formats: plain formats, `Ab32a`, `aBc32b`
* Supported data types: `f32`, `bf16`, `f16`, `s8`, `u8`, `s32`

## Convolution

The implementation supports forward, backward data and backward weights
directions.

* Supported input/output formats: plain formats
* Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`;
* Supported data types: `f32`, `bf16`, `f16`, `s32`, `s8`, `u8`
* Limitations

>     * Some very large problem sizes currently return `unimplemented` due to an

Reviewer: ```suggestion
    * Few substantial problem sizes currently return `unimplemented` due to an
```

      issue with long execution times

## Concat

A generic Concat implementation which uses the Reorder primitive is used. Refer

> to [src/gpu/generic/ref_concat.hpp](../ref_concat.hpp) for more information.

Reviewer 2: I think it's an implementation detail that may not be very useful for the users. Each primitive can have multiple implementations so I believe we just need to document what the primitive supports and not a particular implementation. When it makes sense we can always document peculiarities that can be related to a certain piece of hardware (e.g. A100 matmul, etc).

Author: Good point, added details.

## Deconvolution

The implementation supports forward and backward data and backward weights
directions.

* Supported input/output formats: plain formats

> * Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`;

Reviewer: ```suggestion
* Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`
```

* Supported data types: `f32`, `bf16`, `f16`, `s32`, `s8`, `u8`
* Limitations

>     * Some very large problem sizes currently return `unimplemented` due to an

Reviewer: ```suggestion
    * Few substantial problem sizes currently return `unimplemented` due to an
```

`gelu_tanh`, `hardsigmoid`, `hardswish`, `linear`, `log`, `logistic`, `mish`,
`pow`, `relu`, `round`, `soft_relu`, `sqrt`, `square`,`swish` and `tanh`
* Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`, `NC`, `N`
* Supported data types: `f32`, `bf16`, `f16`, `s32`, `s8`, `u8`

## Layer Normalization

The implementation supports both forward and backward directions.

* Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`, `NC`
* Supported input/output data types for forward direction: `f32`, `bf16`, `f16`,
  `s8`, `u8`
* Supported input/output data types for backward direction: `f32`, `bf16`
* Supported scale/shift data types: `f32`, `bf16`, `f16`

## LRN

The implementation supports both forward and backward directions.

* Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`, `NC`
* Supported data types: `f32`, `bf16`, `f16`

## Matmul

> * Supported formats: plain formats

Reviewer 2: Should we mention something about leveraging the cublaslt format when format tag is any?

Reviewer 2: Nevermind, I confused the documentations for NVIDIA and generic vendors :)

* Supported input/output data types: `f32`, `bf16`, `f16`, `s8`, `u8`, `s32`
* Limitations
    * Runtime dims is not supported
    * PReLU post-op is not supported

## Pooling

The implementation supports both forward and backward directions.

* Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`

> * Supported data types for forward direction: f32, bf16, f16, s8, u8

Reviewer: ```suggestion
* Supported data types for forward direction: `f32`, `bf16`, `f16`, `s8`, `u8`
```

> * Supported data types for backward direction: f32, bf16, f16

Reviewer: ```suggestion
* Supported data types for backward direction: `f32`, `bf16`, `f16`
```
